lib/metrics/events.js

```diff
@@ -172,3 +172,22 @@ function optionallySetService (data, request) {
     (request.query && request.query.service)
 }
 
+function coalesceLocale (data, request) {
+  if (data && data.locale) {
+    return data.locale
+  }
+
+  return request.app && request.app.acceptLanguage
+}
+
+function coalesceUid (data, request) {
+  if (data && data.uid) {
+    return data.uid
+  }
+
+  return request.auth &&
+    request.auth.credentials &&
+    request.auth.credentials.uid &&
+    request.auth.credentials.uid.toString('hex')
```

Comment:
We have stuff like this to find the `uid` in a few places now.  A nice follow up bug would be making all those use just a single function.